numpy/_core/src/multiarray/scalartypes.c.src

```diff
@@ -3904,45 +3904,25 @@ static inline npy_hash_t
  * #lname = datetime, timedelta#
  * #name = Datetime, Timedelta#
  */
-#if NPY_SIZEOF_HASH_T==NPY_SIZEOF_DATETIME
 static npy_hash_t
 @lname@_arrtype_hash(PyObject *obj)
 {
-    npy_hash_t x = (npy_hash_t)(PyArrayScalar_VAL(obj, @name@));
-    if (x == -1) {
-        x = -2;
-    }
-    return x;
-}
-#elif NPY_SIZEOF_LONGLONG==NPY_SIZEOF_DATETIME
-static npy_hash_t
-@lname@_arrtype_hash(PyObject *obj)
-{
-    npy_hash_t y;
-    npy_longlong x = (PyArrayScalar_VAL(obj, @name@));
+    PyArray_DatetimeMetaData *meta;
+    PyArray_Descr *dtype;
+    npy_@lname@ val = PyArrayScalar_VAL(obj, @name@);
 
-    if ((x <= LONG_MAX)) {
-        y = (npy_hash_t) x;
+    if (val == NPY_DATETIME_NAT) {
+        return (*PyBaseObject_Type.tp_hash)(obj);
```

Comment:
```suggestion
        /* Use identity similar to NaN */
        return PyBaseObject_Type.tp_hash(obj);
```
I don't think you need to dereference the function pointer?

Reply:
That's right. It's a style issue. I am trying to fit in with the "numpy/cpython style". See references to `bf_getbuffer` and `bf_releasebuffer` in scalartypes.c.src. Or references to "tp_hash" in the cpython source. They all use (*funcptr)(...).
For what it's worth, in my own personal projects I don't do this, I do what you suggest. :-)

Reply:
Hmmm, very sure most of NumPy doesn't dereference the function pointer...

Reply:
OK, yeah you are right, there are lots of places with the no-dereference style in numpy.
I'll change it. Not a big deal.